src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -863,13 +870,23 @@ public static object DeepCopyInner(object original)
             {
                 copy = copier(original);
                 SerializationContext.Current.RecordObject(original, copy);
+                return copy;
             }
-            else
+
+            var externalSerializer = externalSerializers.FirstOrDefault(serializer => serializer.IsSupportedType(t));
```

Comment:
To clarify the below position
> I don't really have any issue with multiple seralizers, as long as the selection process is explicit and testable. It should not depend on the capabilities of the serializers. My preference for a single serializer was a means of enforcing this strict behavior, but I'm quite open to other suggestions.

My concern is the mechanism of serializer selection.

@Carlm-MS As I understand the current proposal, the serializer selected to serialize a type is the first serializer that supports that type.  Am I understanding this correctly?

@galvesribeiro You wrote
>I just want to make a single semantics for register and use serializers

Can you please go into more details.  I don't disagree, there is just not enough of a concrete suggestion for me to assess.
For instance:
* If we use attributes
  * Would a serializer attribute specify a specific serializer type, or a configured serializer name like the storage provider attribute?
  * Are the attributes on the grain or the data type?
    * If grain, how do we resolve a grain with one serializer calling another grain with a different serializer?
      * Caller always uses the serializer specified by the grain being called?  So serializer must be part of the grain client interface?
    * If on the data type, what happens if multiple types with different serializers are nested in the same serializable object?

I promise I'm not trying to be difficult, I'm just of the opinion that serialization can't be ambiguous.  Suggesting a single serializer was my simplistic attempt to push this problem onto the IExternalSerializer implementers plate. :/